plugins/module_utils/network/sonic/config/network_policy/network_policy.py

```diff
@@ -0,0 +1,392 @@
+
+                        if app_type:
+                            app_type = app_type.upper()
+                            app_dict.update({'type': app_type, 'config': {'type': app_type}})
+                        if vlan_id:
+                            app_dict['config']['vlan-id'] = vlan_id
+                        if tagged is not None:
+                            app_dict['config']['tagged'] = tagged
+                        if priority is not None:
+                            app_dict['config']['priority'] = priority
+                        if dscp is not None:
+                            app_dict['config']['dscp'] = dscp
+                        if app_dict:
+                            app_list.append(app_dict)
+                    if app_list:
+                        policy_dict['applications'] = {'application': app_list}
+                if policy_dict:
+                    policy_list.append(policy_dict)
+            if policy_list:
+                payload = {'openconfig-network-policy-ext:network-policies': {'network-policy': policy_list}}
+                request = {'path': NETWORK_POLICY_PATH, 'method': PATCH, 'data': payload}
+
+        return request
+
+    def get_delete_network_policy_requests(self, commands, is_delete_all):
+        requests = []
+
+        if not commands:
+            return requests
+        if is_delete_all:
+            requests.append({'path': NETWORK_POLICY_PATH, 'method': DELETE})
+            return requests
+
+        for policy in commands:
+            number = policy.get('number')
+            applications = policy.get('applications')
+
+            if number and not applications:
+                requests.append(self.get_delete_network_policy_request(number))
+                continue
+            if applications:
+                for app in applications:
+                    app_type = app.get('app_type')
+                    vlan_id = app.get('vlan_id')
+                    tagged = app.get('tagged')
+                    priority = app.get('priority')
+                    dscp = app.get('dscp')
+
+                    if app_type and len(app) == 1:
+                        requests.append(self.get_delete_network_policy_request(number, app_type))
+                        continue
+                    if vlan_id:
+                        requests.append(self.get_delete_network_policy_request(number, app_type, 'vlan-id'))
+                    if tagged is not None:
+                        requests.append(self.get_delete_network_policy_request(number, app_type, 'tagged'))
+                    if priority is not None:
+                        requests.append(self.get_delete_network_policy_request(number, app_type, 'priority'))
+                    if dscp is not None:
+                        requests.append(self.get_delete_network_policy_request(number, app_type, 'dscp'))
+
+        return requests
+
+    def get_delete_network_policy_request(self, number, app_type=None, attr=None):
```

Comment:
Can be a staticmethod since 'self' is unused.
```suggestion
    @staticmethod
    def get_delete_network_policy_request(number, app_type=None, attr=None):
```